unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor.cs

```diff
@@ -0,0 +1,1344 @@
+                EditorGUI.indentLevel++;
+                RenderAmbientLayer();
+                EditorGUI.indentLevel--;
+            }
+            EditorGUILayout.LabelField("", GUI.skin.horizontalSlider);
+
+            GUILayout.Space(32);
+            showFogLayer = EditorGUILayout.Foldout(showFogLayer, "Fog Layer", true);
+            if (showFogLayer)
+            {
+                EditorGUI.indentLevel++;
+                RenderFogLayer();
+                EditorGUI.indentLevel--;
+            }
+            EditorGUILayout.LabelField("", GUI.skin.horizontalSlider);
+
+            GUILayout.Space(32);
+            showDLLayer = EditorGUILayout.Foldout(showDLLayer, "Directional Light Layer", true);
+            if (showDLLayer)
+            {
+                EditorGUI.indentLevel++;
+                RenderDirectionalLightLayer();
+                EditorGUI.indentLevel--;
+            }
+            EditorGUILayout.LabelField("", GUI.skin.horizontalSlider);
+
+            GUILayout.Space(32);
+
+            // Render Slots
+            RenderSlots();
+
+            // Render Texture Layers
+            //RenderTextureLayers();
+
+            GUILayout.Space(300);
+            EditorGUILayout.EndScrollView();
+            GUILayout.EndArea();
+            if (GUI.changed)
+            {
+                ApplyOnMaterial();
+            }
+        }
+
+        private void Update()
+        {
+            if (selectedConfiguration == null)
+            {
+                return;
+            }
+
+            if (isPaused)
+            {
+                return;
+            }
+
+            timeOfTheDay += Time.deltaTime;
+            timeOfTheDay = Mathf.Clamp(timeOfTheDay, 0.01f, 24);
+
+            ApplyOnMaterial();
+
+            if (timeOfTheDay >= 24)
+            {
+                timeOfTheDay = 0.01f;
+            }
+
+            Repaint();
+        }
+
+        private void EnsureDependencies()
+        {
+            if (selectedConfiguration == null)
+            {
+                UpdateConfigurationsList();
+            }
+
+            UpdateMaterial();
+
+            EditorUtility.SetDirty(selectedConfiguration);
+
+            // Cache directional light reference
+            directionalLight = GameObject.FindObjectsOfType<Light>(true).Where(s => s.type == LightType.Directional).FirstOrDefault();
+        }
+
+        //bool initialized = false;
+        MaterialReferenceContainer.Mat_Layer matLayer = null;
+
+        void InitializeMaterial()
+        {
+
+            matLayer = MaterialReferenceContainer.i.GetMat_LayerForLayers(selectedConfiguration.slots.Count);
+
+            if (matLayer == null)
+            {
+                matLayer = MaterialReferenceContainer.i.materials[0];
+            }
+
+            selectedMat = matLayer.material;
+            selectedConfiguration.ResetMaterial(selectedMat, matLayer.numberOfSlots);
+            RenderSettings.skybox = selectedMat;
+            //initialized = true;
```

Comment:
```suggestion
```

Reply:
Done. Making all these changes in a new commit